ibis/expr/operations/core.py

```diff
@@ -172,6 +172,14 @@ class Binary(Value):
     left: Value
     right: Value
 
+    def __init__(self, left, right, **kwargs):
+        # Dealias left and right, passing through any additional fields
+        if isinstance(left, Alias):
+            left = left.arg
+        if isinstance(right, Alias):
+            right = right.arg
```

Comment:
I'm aware of `unwrap_alias`, but didn't feel like reorging the code to make use of it here. The alias check is only 2 lines of code anyway.

Reply:
What about other operations? This seems like it only solves the problem for binary operations.

Reply:
Oh, because it's not going into the op.

Well, that's definitely something we should fix in `unwrap_aliases`, not here.

Reply:
I think if you replace `unwrap_alias` with this, then it solves the issue without needing to hack:

```python
from ibis.expr.rewrites import p, x

def unwrap_alias(node: ops.Value) -> ops.Value:
    """Unwrap an alias node."""
    return node.replace(p.Alias(x) >> x, filter=ops.Value)
```

Reply:
Good idea! I think applying it in `sqlize` makes more sense than at the `expr/types` level. Pushed something up.